test/e2e/preemption/preemption_test.go

```diff
@@ -0,0 +1,321 @@
+	yunikorn.UpdateConfigMapWrapper(oldConfigMap, "", annotation)
+
+	ginkgo.By("Port-forward the scheduler pod")
+	var err = kClient.PortForwardYkSchedulerPod()
+	Ω(err).NotTo(gomega.HaveOccurred())
+
+	ginkgo.By("create development namespace")
+	ns, err = kClient.CreateNamespace(dev, nil)
+	gomega.Ω(err).NotTo(gomega.HaveOccurred())
+	gomega.Ω(ns.Status.Phase).To(gomega.Equal(v1.NamespaceActive))
+
+	var nodes *v1.NodeList
+	nodes, err = kClient.GetNodes()
+	Ω(err).NotTo(gomega.HaveOccurred())
+	Ω(len(nodes.Items)).NotTo(gomega.BeZero(), "Nodes cant be empty")
+
+	// Extract node allocatable resources
+	for _, node := range nodes.Items {
+		// skip master if it's marked as such
+		node := node
+		if k8s.IsMasterNode(&node) || !k8s.IsComputeNode(&node) {
+			continue
+		}
+		if Worker == "" {
+			Worker = node.Name
+		} else {
+			nodesToTaint = append(nodesToTaint, node.Name)
+		}
+	}
+	Ω(Worker).NotTo(gomega.BeEmpty(), "Worker node not found")
+
+	ginkgo.By("Tainting some nodes..")
+	err = kClient.TaintNodes(nodesToTaint, taintKey, "value", v1.TaintEffectNoSchedule)
+	Ω(err).NotTo(gomega.HaveOccurred())
+
+	nodesDAOInfo, err := restClient.GetNodes(constants.DefaultPartition)
+	Ω(err).NotTo(gomega.HaveOccurred())
+	Ω(nodesDAOInfo).NotTo(gomega.BeNil())
+
+	for _, node := range *nodesDAOInfo {
+		if node.NodeID == Worker {
+			WorkerMemRes = node.Available["memory"]
+		}
+	}
+	WorkerMemRes /= (1000 * 1000) // change to M
+	fmt.Fprintf(ginkgo.GinkgoWriter, "Worker node %s available memory %dM\n", Worker, WorkerMemRes)
+
+	sleepPodMemLimit = int64(float64(WorkerMemRes) / 3)
+	Ω(sleepPodMemLimit).NotTo(gomega.BeZero(), "Sleep pod memory limit cannot be zero")
+	fmt.Fprintf(ginkgo.GinkgoWriter, "Sleep pod limit memory %dM\n", sleepPodMemLimit)
+})
+
+var _ = ginkgo.AfterSuite(func() {
+
+	ginkgo.By("Untainting some nodes")
+	err := kClient.UntaintNodes(nodesToTaint, taintKey)
+	Ω(err).NotTo(gomega.HaveOccurred(), "Could not remove taint from nodes "+strings.Join(nodesToTaint, ","))
+
+	ginkgo.By("Check Yunikorn's health")
+	checks, err := yunikorn.GetFailedHealthChecks()
+	Ω(err).NotTo(gomega.HaveOccurred())
+	Ω(checks).To(gomega.Equal(""), checks)
+
+	testDescription := ginkgo.CurrentSpecReport()
+	if testDescription.Failed() {
+		tests.LogTestClusterInfoWrapper(testDescription.FailureMessage(), []string{ns.Name})
+		tests.LogYunikornContainer(testDescription.FailureMessage())
+	}
+	ginkgo.By("Tearing down namespace: " + ns.Name)
+	err = kClient.TearDownNamespace(ns.Name)
+	Ω(err).NotTo(gomega.HaveOccurred())
+
+	yunikorn.RestoreConfigMapWrapper(oldConfigMap, annotation)
+})
+
+var _ = ginkgo.Describe("Preemption", func() {
+	ginkgo.It("Verify_basic_preemption", func() {
+		ginkgo.By("A queue uses resource more than the guaranteed value even after removing one of the pods. The cluster doesn't have enough resource to deploy a pod in another queue which uses resource less than the guaranteed value.")
+		// update config
```

Comment:
Either update config in BeforeSuite and reset in AfterSuite, or do it BeforeEach/AfterEach. If you want to do this in every test case, then perform a reset step also.

Right now, it's problematic because you don't perform restoration. If the config is the same for every test, then I suggest BeforeSuite/AfterSuite. There are examples for config saving/restoration in different suites, check those out.

Reply:
Updated it to use `UpdateCustomConfigMapWrapper` in each test case and use `RestoreConfigMapWrapper` in AfterEach.